.github/workflows/publish-to-pypi.yml

```diff
@@ -0,0 +1,176 @@
+name: Publish and Test
+
+on:
+  push:
+    tags:
+      - '*'
+
+env:
+  VERSION: (echo '${{ github.ref }}' | sed -e 's,.*/\(.*\),\1,' | sed -e 's/^v//')
+
+jobs:
+  build-n-publish-test:
+    name: Build and publish to TestPyPI
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ '3.7', '3.8', '3.9', '3.10', '3.11' ]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4.4.0
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install pypa/build
+        run: python -m pip install build --user
+      - name: Build a binary wheel and a source tarball
+        run: python -m build --sdist --wheel --outdir dist/ .
+      - name: Save built package
+        uses: actions/upload-artifact@v3
+        with:
+          name: dist
+          path: dist/
+          retention-days: 1
+      - name: Publish to Test PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          skip_existing: true
+          password: ${{ secrets.TEST_PYPI_API_TOKEN }}
+          repository_url: https://test.pypi.org/legacy/
+  unit-tests-test:
+    name: Running unit tests for TestPyPI
+    needs: build-n-publish-test
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ '3.7', '3.8', '3.9', '3.10', '3.11' ]
+        airflow-version: [ '2.0.2', '2.2.5', '2.3.4', '2.4.3', '2.5.0' ]
+    steps:
+      - uses: actions/checkout@v3
+      - run: rm -rf airflow_clickhouse_plugin
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4.4.0
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install -r requirements.txt --ignore-installed apache-airflow==${{ matrix.airflow-version }}
```

Comment:
this is a great point, because I'm looking at matrices right now, and I don't understand how unit tests pass, given that for example 2.0.2 should not work with python 11 (https://github.com/apache/airflow/tree/ 2.0.2#requirements).

@bryzgaloff matrices were supposed to work like matrices, or like zip?

Reply:
"Being tested" does not mean "not working if not tested": py3.11 + Airflow 2.0.2 can still be a working combination, but not stated to be tested and intended to be maintained.

I expect the matrices to work as cartesian join, _not_ a zip, right.